### ISSUE TITLE
The names of user contribution docsets can't be fully displayed

### ISSUE THREAD

**Issue by integzz** [zealdocs/zeal#1240]
The names of user contribution docs can't be fully displayed and the folder names of these docsets are not right, either. 

Since I can not upload an image, you may need to try to add a feed URL from below link to see

[zeal user contributions](https://zealusercontributions.herokuapp.com/)

**Comment by nullptr-leo** [comment]
> The names of user contribution docs can't be fully displayed and the folder names of these docsets are not right, either.
> 
> Since I can not upload an image, you may need to try to add a feed URL from below link to see
> 
> [zeal user contributions](https://zealusercontributions.herokuapp.com/)

Yes, the last 4 characters are always truncatesd. It seems zeal regard them as `.xml`.

```
http://zealusercontributions.herokuapp.com/api/docsets/PyTorch -> PyT
http://zealusercontributions.herokuapp.com/docsets/PyTorch.xml -> PyTorch
```

**Comment by trollixx** [comment]
I added a quick check for the extension.

Zeal should not rely on the name of the file as there could be none. That will be addressed in a more involved fix for #514.
